File: sample_backbone_app/src/main/java/org/researchstack/backboneapp/AppPrefs.java
package org.researchstack.backboneapp;
import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;


public class AppPrefs
{
    public static final String HAS_CONSENTED     = "HAS_CONSENTED";
    public static final String CONSENT_NAME      = "CONSENT_NAME";
    public static final String CONSENT_SIGNATURE = "CONSENT_SIGNATURE";
    public static final String HAS_SURVEYED      = "HAS_SURVEYED";
    public static final String SURVEY_RESULT     = "SURVEY_RESULT";

    //-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
    // Statics
    //-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
    private static AppPrefs instance;

    //-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
    // Field Vars
    //-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
    private final SharedPreferences prefs;

    AppPrefs(Context context)
    {
        prefs = PreferenceManager.getDefaultSharedPreferences(context);
    }

    public static synchronized AppPrefs getInstance(Context context)
    {
        if(instance == null)
        {
            instance = new AppPrefs(context);
        }
        return instance;
    }

    public boolean hasConsented()
    {
        return prefs.getBoolean(HAS_CONSENTED, false);
    }

    public void setHasConsented(boolean consented)
    {
        prefs.edit().putBoolean(HAS_CONSENTED, consented).apply();
    }

    public boolean hasSurveyed()
    {
        return prefs.getBoolean(HAS_SURVEYED, false);
    }

    public void setHasSurveyed(boolean surveyed)
    {
        prefs.edit().putBoolean(HAS_SURVEYED, surveyed).apply();
    }
}

File: foundationexampleapp/src/main/java/org/researchstack/foundationexampleapp/AppPrefs.kt
package org.researchstack.kotlinbackbonesampleapp

import android.content.Context
import android.content.SharedPreferences
import android.preference.PreferenceManager

class AppPrefs {

    //-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
    // Statics
    //-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=

    companion object {

        val HAS_CONSENTED = "HAS_CONSENTED"
        val CONSENT_NAME = "CONSENT_NAME"
        val CONSENT_SIGNATURE = "CONSENT_SIGNATURE"
        val HAS_SURVEYED = "HAS_SURVEYED"
        val SURVEY_RESULT = "SURVEY_RESULT"

        private var instance: AppPrefs? = null

        @Synchronized
        fun getInstance(context: Context): AppPrefs {
            return this.instance?.also { a ->
                return a
            } ?: run {
                val instance = AppPrefs(context)
                this.instance = instance
                return instance
            }

        }
    }

    //-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
    // Field Vars
    //-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
    private var prefs: SharedPreferences


    constructor(context: Context) {
        this.prefs = PreferenceManager.getDefaultSharedPreferences(context)
    }

    fun hasConsented(): Boolean {
        return this.prefs.getBoolean(HAS_CONSENTED, false)
    }

    fun setHasConsented(consented: Boolean) {
        this.prefs.edit().putBoolean(HAS_CONSENTED, consented).apply()
    }

    fun hasSurveyed(): Boolean {
        return this.prefs.getBoolean(HAS_SURVEYED, false)
    }

    fun setHasSurveyed(surveyed: Boolean) {
        this.prefs.edit().putBoolean(HAS_SURVEYED, surveyed).apply()
    }
}